logwatch/logwatch.go

```diff
@@ -0,0 +1,181 @@
+package logwatch
+
+import (
+	"context"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/prometheus/common/model"
+	"github.com/rs/zerolog"
+	"github.com/smartcontractkit/wasp"
+	"github.com/testcontainers/testcontainers-go"
+
+	"github.com/smartcontractkit/chainlink-testing-framework/utils"
+)
+
+// LogNotification notification about log line match for some container
+type LogNotification struct {
+	Container string
+	Log       string
+}
+
+// LogWatch is a test helper struct to monitor docker container logs for some patterns
+// and push their logs into Loki for further analysis
+type LogWatch struct {
+	t          *testing.T
+	log        zerolog.Logger
+	loki       *wasp.LokiClient
+	patterns   map[string][]*regexp.Regexp
+	notifyTest chan *LogNotification
+	containers []testcontainers.Container
+	consumers  map[string]*ContainerLogConsumer
+}
+
+// NewLogWatch creates a new LogWatch instance, with a Loki client
+func NewLogWatch(t *testing.T, patterns map[string][]*regexp.Regexp) (*LogWatch, error) {
+	loki, err := wasp.NewLokiClient(wasp.NewEnvLokiConfig())
```

Comment:
Would it make sense to move the loki client into the chainlink-env code base since it is part of our env stuff? Would also mean we don't need to pull in wasp for everything that uses chainlink-env.

Reply:
We can move it later since we remove chainlink-env in the future.
Also, I'm using it from `wasp` just because of it's `vendored`. Just to not cause more dep hell.